src/inject_key_bin.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+
+#
+# Use this script to inject your own private key and authentication counter
+# into U2F binary. Might be useful if you want keys to survive firmware updates.
+#
+# Example:
+#
+# Generate EC private key with openssl:
+# > openssl ecparam -name prime256v1 -genkey -noout -outform der > key.der
+#
+# Inject generated key into u2f.bin and set auth counter to 100:
+# > python3 inject_key_bin.py --key key.der --ctr 100
+#
+# key will not be modified if --key parameter is not present
+# counter will not be modified if --ctr parameter is not present
+
+from __future__ import print_function
+from asn1crypto.keys import ECPrivateKey
+import hashlib
+import argparse
+import sys
+import struct
+import os
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--bin", default="build/u2f.bin",
+                    help='.bin file to inject keys into. Or "stdin"')
+parser.add_argument("--key", help="EC private key in DER format")
+parser.add_argument("--ctr", default=0, type=lambda x: int(x,0), help="Value of auth counter")
+parser.add_argument("--offset", default=0xB800, type=lambda x: int(x,0), help="Offset within file to patch")
```

Comment:
Maybe we could guess by looking at the .bin file whether it is Tomu binary of stm32f1x? Or at least we could have two preset values one for each of them. Also I don't remember if playing with make flags can change the offset.

Reply:
All builds share same .ld file, so I think offsets are really constant.  Maybe this argument can be removed in favour of a constant.  In a pinch, a user can edit the .py to change the value?

Reply:
I'm wrong about all builds sharing same .ld.  There is also efm32hg.ld